docs/src/pages/power-profiler.js

```diff
@@ -0,0 +1,357 @@
+  const { value: glowBrightness, bind: bindGlowBrightness } = useInput(1);
+
+  const { value: displayEnabled, bind: bindDisplayEnabled } = useInput(false);
+  const { value: displayType, bind: bindDisplayType } = useInput("");
+
+  const currentBoard =
+    board === "custom"
+      ? {
+          powerSupply: {
+            type: psuType,
+            outputVoltage: outputV,
+            quiescent,
+            efficiency,
+          },
+          otherQuiescent,
+        }
+      : zmkBoards[board];
+
+  return (
+    <Layout
+      title={`ZMK Power Profiler`}
+      description="Estimate your keyboard's power usage and battery life on ZMK."
+    >
+      <header className={classnames("hero hero--primary", styles.heroBanner)}>
+        <div className="container">
+          <h1 className="hero__title">ZMK Power Profiler</h1>
+          <p className="hero__subtitle">
+            {"Estimate your keyboard's power usage and battery life on ZMK."}
+          </p>
+        </div>
+      </header>
+      <main>
+        <section className="container">
+          <div className="profilerSection">
+            <h3>Keyboard Specifications</h3>
+            <div className="row">
+              <div className="col col--4">
+                <div className="profilerInput">
+                  <label>Board</label>
+                  <select value={board} {...bindBoard}>
+                    <option hidden value="">
+                      Select a board
+                    </option>
+                    {Object.keys(zmkBoards).map((b) => (
+                      <option key={b}>{b}</option>
+                    ))}
+                    <option value="custom">Custom</option>
+                  </select>
+                </div>
+              </div>
+              <div className="col col--4">
+                <div className="profilerInput">
+                  <label>Split Keyboard</label>
+                  <input
+                    id="split"
+                    checked={split}
+                    {...bindSplit}
+                    className="toggleInput"
+                    type="checkbox"
+                  />
+                  <label htmlFor="split" className="toggle">
+                    <div className="toggleThumb" />
+                  </label>
+                </div>
+              </div>
+              <div className="col col--4">
+                <div className="profilerInput">
+                  <label>Battery Size</label>
+                  <div className="inputBox">
+                    <input
+                      value={batterySize}
+                      {...bindBatterySize}
+                      type="number"
+                    />
+                    <span>mAh</span>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          {board === "custom" && (
+            <div className="profilerSection">
```

Comment:
Consider pulling this massive block out into a separate component and using something like `useContext()` to share its state back to this component.

Reply:
Agreed, breaking this into subcomponents might help readability - my eyes glossed over on this file.  That said, I appreciate that'll probably increase the boilerplate code somewhat.